pkg/cluster/resources.go

```diff
@@ -65,6 +65,17 @@ func (c *Cluster) listResources() error {
 
 func (c *Cluster) createStatefulSet() (*appsv1.StatefulSet, error) {
 	c.setProcessName("creating statefulset")
+	// check if it's allowed that spec contains initContainers
+	if c.Spec.InitContainers != nil && len(c.Spec.InitContainers) > 0 &&
```

Comment:
Does  the validation reject the manifest with empty  `initContainers` list  ?  that looks like a misconfiguration at first glance. Same for sidecar containers.

Reply:
validation does not reject empty arrays